src/protocols/secure_channel/PairingSession.h

```diff
@@ -132,31 +132,20 @@ class DLL_EXPORT PairingSession : public SessionDelegate
         return CHIP_ERROR_INTERNAL;
     }
 
+    /*
+     * Helper function to send a status report. This function sends a status report without any protocol data.
+     */
     void SendStatusReport(Messaging::ExchangeContext * exchangeCtxt, uint16_t protocolCode)
     {
+        VerifyOrReturn(exchangeCtxt);
+
         Protocols::SecureChannel::GeneralStatusCode generalCode = (protocolCode == Protocols::SecureChannel::kProtocolCodeSuccess)
             ? Protocols::SecureChannel::GeneralStatusCode::kSuccess
             : Protocols::SecureChannel::GeneralStatusCode::kFailure;
 
-        ChipLogDetail(SecureChannel, "Sending status report. Protocol code %d, exchange %d", protocolCode,
-                      exchangeCtxt->GetExchangeId());
-
         Protocols::SecureChannel::StatusReport statusReport(generalCode, Protocols::SecureChannel::Id, protocolCode);
 
-        auto handle = System::PacketBufferHandle::New(statusReport.Size());
-        VerifyOrReturn(!handle.IsNull(), ChipLogError(SecureChannel, "Failed to allocate status report message"));
-        Encoding::LittleEndian::PacketBufferWriter bbuf(std::move(handle));
-
-        statusReport.WriteToBuffer(bbuf);
-
-        System::PacketBufferHandle msg = bbuf.Finalize();
-        VerifyOrReturn(!msg.IsNull(), ChipLogError(SecureChannel, "Failed to allocate status report message"));
-
-        CHIP_ERROR err = exchangeCtxt->SendMessage(Protocols::SecureChannel::MsgType::StatusReport, std::move(msg));
-        if (err != CHIP_NO_ERROR)
-        {
-            ChipLogError(SecureChannel, "Failed to send status report message: %" CHIP_ERROR_FORMAT, err.Format());
-        }
+        exchangeCtxt->SendStatusReport(statusReport);
```

Comment:
would it make sense to log errors here instead of having them inside SendStatusReport and then ignoring the returned CHIP_ERROR?

Reply:
+1. I think it's best to bubble the error to the highest level and then take the necessary action, be it logging or otherwise.